src/main/java/com/hitesh/TicTacToe/model/Board.java

```diff
@@ -0,0 +1,62 @@
+package com.hitesh.TicTacToe.model;
+
+import com.hitesh.TicTacToe.constant.Constants;
+import lombok.Getter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Getter
+public class Board {
+
+    private String id;
+    private List<List<Cell>> cellList;
+
+    public Board(String id, List<List<Cell>> cellList) {
+        this.id = id;
+        this.cellList = cellList;
+        initialize();
+    }
+
+    private void initialize() {
+
+        for(int i=0;i<Constants.NO_OF_ROWS;i++){
+
+            List<Cell> newCells = new ArrayList<>();
+
+            for(int j=0;j<Constants.NO_OF_COLS;j++){
+                newCells.add(new Cell(i,j, null));
+            }
+
+            cellList.add(newCells);
+        }
+    }
+
+    boolean isCellOccupied(Cell cell){
+
+        return cellList.get(cell.getX()).get(cell.getY()).getPiece() != null;
+    }
+
+    public void assignPieceToCell(Cell cell, Piece piece) {
+        cellList.get(cell.getX()).get(cell.getY()).assignPiece(piece);
+    }
+
+    public void print(){
+
+        for(int i=0;i<Constants.NO_OF_ROWS;i++){
+            for(int j=0;j<Constants.NO_OF_COLS;j++){
```

Comment:
Should it make more sense to set size of rows and cols by the application class itself rather than using constant? Two reasons: 
1. Dependency injection will be better for testing 
2. Application should have logic(indirectly user has it) to change sizes